components/vfs/src/root_file_system.cpp

```diff
@@ -256,19 +274,16 @@ StackErrorPtr RootFileSystem::enumerate_folders_recursive(const std::string &fil
 			return res;
 		}
 
-		for (std::string &dir : dirs)
+		for (auto &dir : dirs)
 		{
-			std::string full_path = front + dir;
-			auto        it        = all_dirs.find(full_path);
-			if (it == all_dirs.end())
-			{
-				all_dirs.emplace(full_path);
-				dirs_to_visit.emplace(full_path);
-			}
+			dirs_to_visit.push(dir);
 		}
+
```

Comment:
One might think about changing `dirs_to_visit` from `std::queue` to `std::deque`, allowing to insert all the elements from `dirs` at the end of `dirs_to_visit` with one call?

Reply:
Good call!